### PR TITLE
PTX-15473 [TC Automation] Add NetworkDelayWhilePoolExpand test case

### PR DESCRIPTION
**Pull request by spai-px**
**What this PR does / why we need it**:
Adds test case NetworkDelayWhilePoolExpand 
**Which issue(s) this PR fixes** (optional)
Closes # PTX-15473

**Special notes for your reviewer**:
[Aetos link](https://aetos.pwx.purestorage.com/resultSet/testSetID/581991 )

### CODE REVIEW

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -11356,3 +11356,105 @@ var _ = Describe("{driveAddPxRestart}", func() {
...
+				err = waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isJournalEnabled)
+				log.FailOnError(err, "Failed to wait for pool to be resized")
+
+				poolToResize = getStoragePool(selectedPool.Uuid)
+				expectedSize = selectedPool.TotalSize/units.GiB + 150
+
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, expectedSize, true)
+				log.FailOnError(err, "Failed to expand pool with add disk")
+				err = waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isJournalEnabled)
+				log.FailOnError(err, "Failed to wait for pool to be resized")
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:11432]: How about we log node and pool details?

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -11356,3 +11356,105 @@ var _ = Describe("{driveAddPxRestart}", func() {
...
+
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, true)
+				log.FailOnError(err, "Failed to expand pool with add disk")
+
+				err = waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isJournalEnabled)
+				log.FailOnError(err, "Failed to wait for pool to be resized")
+
+				poolToResize = getStoragePool(selectedPool.Uuid)
+				expectedSize = selectedPool.TotalSize/units.GiB + 150
+
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:11428]: How about increasing the timeout, given that there is a network delay?

File: tests/basic/storage_pool_test.go [hunk 3]
```diff
@@ -11356,3 +11356,105 @@ var _ = Describe("{driveAddPxRestart}", func() {
...
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, expectedSize, true)
```
Comment (kphalgun-px) [inline tests/basic/storage_pool_test.go:11429]: How about increasing the timeout, given that there is a network delay?

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

File: tests/basic/storage_pool_test.go (hunk 3 above)
Comment (spai-px) [reply tests/basic/storage_pool_test.go:11429]: Cannot manipulate the timout so we will see if it fails in the future.

File: tests/basic/storage_pool_test.go (hunk 1 above)
Comment (spai-px) [reply tests/basic/storage_pool_test.go:11432]: Done.

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Pull request merged**